app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/address/Postcode.scala

```diff
@@ -26,4 +29,21 @@ object Postcode {
   implicit val format: Format[Postcode] =
     implicitly[Format[String]].inmap(Postcode(_), _.value)
 
+  val mapping: Mapping[Postcode] = {
+    val postcodeRegexPredicate =
+      "^[A-Z]{1,2}[0-9][0-9A-Z]?\\s?[0-9][A-Z]{2}$|BFPO\\s?[0-9]{1,3}$".r.pattern
+        .asPredicate()
+
+    def validatePostcode(p: Postcode): ValidationResult = {
+      val postcodeWithoutSpaces = p.value.toUpperCase.replaceAllLiterally(" ", "")
```

Comment:
we need to get the rules with stripping/no stripping out spaces defined at some point